pysquared/state_of_health.py

```diff
@@ -0,0 +1,174 @@
+from collections import OrderedDict
+
+import microcontroller
+
+from pysquared.logger import Logger
+from pysquared.protos.imu import IMUProto
+from pysquared.protos.power_monitor import PowerMonitorProto
+from pysquared.protos.radio import RadioProto
+
+try:
+    from typing import Any, OrderedDict
+
+    from .nvm.counter import Counter
+    from .nvm.flag import Flag
+
+except Exception:
+    pass
+
+
+class StateOfHealth:
+    def __init__(
+        self,
+        logger: Logger,
+        battery_power_monitor: PowerMonitorProto,
+        solar_power_monitor: PowerMonitorProto,
+        radio_manager: RadioProto,
+        imu_manager: IMUProto,
+        boot_count: Counter,
+        burned_flag: Flag,
+        brownout_flag: Flag,
+        fsk_flag: Flag,
+    ) -> None:
+        self.logger: Logger = logger
+        self.battery_power_monitor: PowerMonitorProto = battery_power_monitor
+        self.solar_power_monitor: PowerMonitorProto = solar_power_monitor
+        self.radio_manager: RadioProto = radio_manager
+        self.imu_manager: IMUProto = imu_manager
+        self.boot_count: Counter = boot_count
+        self.burned_flag: Flag = burned_flag
+        self.brownout_flag: Flag = brownout_flag
+        self.fsk_flag: Flag = fsk_flag
+
+        self.state: OrderedDict[str, Any] = OrderedDict(
+            [
+                # init all values in ordered dict to None
+                ("system_voltage", None),
+                ("system_current", None),
+                ("solar_voltage", None),
+                ("solar_current", None),
+                ("battery_voltage", None),
+                ("radio_temperature", None),
+                ("radio_modulation", None),
+                ("microcontroller_temperature", None),
+                ("internal_temperature", None),
+                ("error_count", None),
+                ("boot_count", None),
+                ("burned_flag", None),
+                ("brownout_flag", None),
+                ("fsk_flag", None),
+            ]
+        )
+
+    def update(self):
+        """
+        Update the state of health
+        """
+        try:
```

Comment:
Looking over the calls in this try/except I think that none of these calls have the possibility of raising an exception. What are you seeing?